### PR TITLE
Fix supplier statement request model 

### PR DESCRIPTION
**Pull request by aliveni**
### Change description ###
~~The reconciliation endpoint should return `BAD REQUEST` when the supplier statement contains an invalid field.~~

**UPDATE**
The reconciliation request model contains an additional field `report` which is not in the spec. 
- Updated the request model 
- Updated the payload used in the tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/resources/reconciliation/invalid-supplier-statement-report.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
-  "envelopes": {
-    "zip_file_name": "2010404021234_14-08-2020-03-06-18.zip",
-    "container": "bulkscan",
-    "jurisdiction": "BULKSCAN"
+  "report": {
+    "envelope-list": [
```
Comment (aliveni) [inline src/integrationTest/resources/reconciliation/invalid-supplier-statement-report.json:3]: invalid field name

**Review by mdayican — APPROVED** [review]

**Review by nitinprabhu — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/model/in/SupplierStatement.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
     @NotNull
     public final List<Envelope> envelopes;
 
-    public SupplierStatement(@JsonProperty("envelopes") List<Envelope> envelopes) {
+    public SupplierStatement(@JsonProperty(value = "envelopes", required = true) List<Envelope> envelopes) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/model/in/SupplierStatement.java:15]: Can this be set in an annotation on the field instead?
There is one already (`@NotNull`)

File: src/integrationTest/resources/reconciliation/invalid-supplier-statement-report.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
-  "envelopes": {
-    "zip_file_name": "2010404021234_14-08-2020-03-06-18.zip",
-    "container": "bulkscan",
-    "jurisdiction": "BULKSCAN"
+  "report": {
```
Comment (rkondratowicz) [inline src/integrationTest/resources/reconciliation/invalid-supplier-statement-report.json:2]: not related to this PR, but is this field in the spec??

Comment (aliveni) [reply src/integrationTest/resources/reconciliation/invalid-supplier-statement-report.json:2]: I remember copying it from the samples in reconciliation design proposals, but I don't see it now. I'll raise a pr to remove `report`.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/model/in/SupplierStatement.java (hunk 2 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/model/in/SupplierStatement.java:15]: This is no longer required. it validates with `@NotNull` annotation on the field

File: src/integrationTest/resources/reconciliation/invalid-supplier-statement-report.json (hunk 3 above)
Comment (aliveni) [reply src/integrationTest/resources/reconciliation/invalid-supplier-statement-report.json]: Fixed the request model to match with the spec

**Review by rkondratowicz — APPROVED** [review]

**Pull request merged**